Fix: Columnar handling of duplicate column values

thekaveman opened the pull request:
Closes #13.

## The bug

This was a tricky one! But a silly mistake in retrospect. The code that transforms columns into rows previously looked like (pseudocode):

```csharp
// a list of lists of the row values
rows = [[],[],[],[]]

foreach (column in columns)
{
    foreach (value in column)
    {
        rows[column.index(value)].add(value)
    }
}
```

So finding the index of `value` in the `column`, and using that to add the `value` to the end of the corresponding row list. 

Finding the index only gets the index of the first occurrence of the `value`! So when `column` contains duplicates, subsequent `value`s are appended to the same row list over and over.

## The fix

This was a really simple fix, just iterating over the column indices directly:

```csharp
rows = [[],[],[],[]]

foreach (column in columns)
{
    for (i = 0; i < column.count; i++)
    {
        rows[i].add(column[i])
    }
}
```

Tests have been added to ensure the correct behavior for duplicate column values.

angela-tran commented on tests/Data/Csv/ColumnarTests.cs:
```diff
@@ -220,5 +276,70 @@ public void Remove_DoesNotRemove_EmptyOrNonExistentColumns(string[] columns)
+        }
+
+        [TestMethod]
+        public void Rows_MultiRow_Duplicates()
```
Nice test 👍

angela-tran commented on tests/Data/Csv/ColumnarTests.cs:
```diff
@@ -220,5 +276,70 @@ public void Remove_DoesNotRemove_EmptyOrNonExistentColumns(string[] columns)
+        [TestMethod]
+        public void Rows_MultiRow()
+        {
+            var columner = NewColumnar(_data);
```
This typo doesn't break anything, but easy to fix

angela-tran reviewed (changes requested):
This fix looks good to me! 🎉 I noticed a small typo in a variable name, so after that is fixed, I say this is ready to be merged in.

thekaveman commented:
Addressed your comments @angela-tran, thanks for the catch!

Added some more doc comments, and fixed another small bug with `Columnar.Write()`: it was always using `,` as a delimiter, even for data that was delimited by something else to begin with.

Rebased on the latest `main`.

angela-tran reviewed (approved):
The new changes look good to me 👍
